### PR TITLE
Implement 'OS::shell_show_in_explorer()' for FilesystemDock and expose to GDScript.

### PR DESCRIPTION
**Pull request by Daylily-Zeleen**
Implement 'OS::shell_show_in_explore()' for FilesystemDock and expose to GDScript.

Currently only implement for Windows, because I only have a windows device.
On other platforms, it will fallback to origin action.

I hope someone can implement for other platforms.

https://user-images.githubusercontent.com/61624558/190620562-66997ebe-1ba5-4b18-81f0-7f796c02d838.mp4

Because the has a merge conflict, I remake this pr.
Replace [#65906](https://github.com/godotengine/godot/pull/65906).

### CODE REVIEW

File: platform/windows/os_windows.cpp [hunk 1]
```diff
@@ -824,6 +824,33 @@ Error OS_Windows::shell_open(String p_uri) {
 	}
 }
 
+Error OS_Windows::shell_show_in_explorer(String p_fpath) {
+	if (!p_fpath.begins_with("file://")) {
+		p_fpath = String("file://") + p_fpath;
+	}
+	INT_PTR ret = (INT_PTR)ShellExecuteW(nullptr, nullptr, LPCWSTR(String("explorer.exe").utf16().get_data()), LPCWSTR((String("/select,") + p_fpath).utf16().get_data()), nullptr, SW_SHOWNORMAL);
```
Comment (bruvzg) [inline platform/windows/os_windows.cpp:831]: `String("/select,") + p_fpath`, is it working if path contains spaces? If not, it might need escaping.

Comment (Daylily-Zeleen) [reply platform/windows/os_windows.cpp:831]: Ok,  I use `String("file://") + p_fpath.strip_escapes().strip_edges()` to replace  it. I am not sure it is necessary that using `strip_escapes()` at here.

**Comment by Daylily-Zeleen** [comment]
Can someone help me check abount what's wrong in linux builds?

**Comment by Daylily-Zeleen** [comment]
Replace by [#69698](https://github.com/godotengine/godot/pull/69698)

**Pull request closed without merging**